lib/ui/painting.dart

```diff
@@ -4302,6 +4302,185 @@ class Canvas extends NativeFieldWrapperClass2 {
 
   /// Draws the given [Image] into the canvas with its top-left corner at the
   /// given [Offset]. The image is composited into the canvas using the given [Paint].
+  ///
+  /// {@tool snippet}
+  ///
+  /// An example usage to draw an asset:
```

Comment:
Might be good to put a bit more detail here explaining what the limitations of this approach are, explaining that it's specific to when you're using widgets, and so on